src/main/java/io/jenkins/plugins/ml/utils/ConvertHelper.java

```diff
@@ -0,0 +1,79 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import hudson.FilePath;
+import org.apache.zeppelin.jupyter.JupyterUtil;
+import org.apache.zeppelin.jupyter.zformat.Note;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+
+
+public class ConvertHelper {
+
+    /**
+     * @param jupyterFile Path for the jupyter notebook
+     * @return Plain text which contains python code only
+     * @throws IOException          when file path does not exist
+     * @throws InterruptedException exception on input stream reading
+     */
+    public static String jupyterToText(FilePath jupyterFile) throws IOException, InterruptedException {
+        // Convert Jupyter Notebook to JSON
+        Note n = new JupyterUtil().getNote(new InputStreamReader(jupyterFile.read(), Charset.forName("UTF-8")), "%python", "\n", "#");
+        Gson gson = new Gson();
+        JsonElement tree = gson.toJsonTree(n);
+        JsonObject obj = tree.getAsJsonObject();
+        JsonArray array = obj.get("paragraphs").getAsJsonArray();
+        StringBuilder outText = new StringBuilder();
+        for (JsonElement element : array)
+            if (element.isJsonObject()) {
+                JsonObject cell = element.getAsJsonObject();
+                String code = cell.get("text").getAsString();
+                outText.append(code);
+            }
+        return outText.toString();
+    }
+
+    /**
+     * @param jupyterFile Path for the jupyter notebook
+     * @return JSON object
+     * @throws IOException          when file path does not exist
+     * @throws InterruptedException exception on input stream reading
+     */
+    public static JsonObject jupyterToJSON(FilePath jupyterFile) throws IOException, InterruptedException {
+
+        Note n = new JupyterUtil().getNote(new InputStreamReader(jupyterFile.read(), Charset.forName("UTF-8")), "%python", "\n", "#");
```

Comment:
Q1/ I think the `InputStreamReader` are not being properly closed?

Q2/ I haven't thought about that until now @Loghijiaha , but is the plug-in supposed to work on Windows too? If so, instead of `\n` we can use.. `System.fileSeparator`? Or is it under `File`? Anyway, just in case you think users could be testing it under other OS's.

Q3/ should "%python" be a constant? And by the way,

Q4/ does "%python" mean that we are parsing only Python code from the notebook? Or does it support if I have a notebook with JS code like https://github.com/kinow/notebooks/blob/master/javascript/enumify/Enumify.ipynb?

Reply:
Ans 1:  I will change with the try with resource code block
Ans 2:
It is not related to OS where we run the code. Zeppelin-python API will identify `/n` as a new line to be interpreted next. `/n` is about a new line. not related to a file path. 

Ans 3:

```
Using the Python Interpreter

In a paragraph, use %python to select the Python interpreter and then input all commands.

The interpreter can only work if you already have python installed (the interpreter doesn't bring it own python binaries).
Normally we are activating Zeppelin notebook to python-mode; that means next code will be python code
There are several interpreters like %spark.r, %sql and %pyspark
```
It is not necessary here. It is just an `id` for the note that we try to convert. 
The id will be shown in the top of the result log as `Note converted from Jupyiter_<id>`

Ans 4:
I never tried a JS in Jupiter. But this will not interpret the js. But as usual, this accepts bash commands in this code similar to the Jupiter notebook.